src/notebook/notebook/widget.ts

```diff
@@ -739,32 +743,29 @@ class Notebook extends StaticNotebook {
    */
   protected onUpdateRequest(msg: Message): void {
     // Set the appropriate classes on the cells.
-    let layout = this.layout as PanelLayout;
-    let widget = layout.childAt(this.activeCellIndex) as BaseCellWidget;
+    let activeCell = this.activeCell;
     if (this.mode === 'edit') {
       this.addClass(EDIT_CLASS);
       this.removeClass(COMMAND_CLASS);
-      if (widget) {
-        widget.focus();
-        if (widget instanceof MarkdownCellWidget) {
-          (widget as MarkdownCellWidget).rendered = false;
+      if (activeCell) {
+        activeCell.focus();
+        if (activeCell instanceof MarkdownCellWidget) {
+          (activeCell as MarkdownCellWidget).rendered = false;
```

Comment:
this cast shouldn't be needed. `instanceof` is a type guard, so `activeCell` should be properly typed within the `if` block.